### PR TITLE
feat(console,core): hide admin user

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Hide admin user in AC:

1. In user management table list, admin users are filtered.
2. In audit log detail, if is admin user, the "user" link is removed and a tag is added.
3. Add a tag on special "Admin Console" app.

## Linear Issue Reference

LOG-3041

## Testing

截屏2022-06-21 下午4 13 58

截屏2022-06-21 下午4 18 11

### CODE REVIEW

File: packages/core/src/queries/user.ts [hunk 1]
```diff
@@ -98,15 +98,17 @@ export const countUsers = async (search?: string) =>
   envSet.pool.one<{ count: number }>(sql`
     select count(*)
     from ${table}
-    ${conditionalSql(search, (search) => sql`where ${buildUserSearchConditionSql(search)}`)}
+    where not (${fields.roleNames}::jsonb?${UserRole.Admin})
+    ${conditionalSql(search, (search) => sql`and ${buildUserSearchConditionSql(search)}`)}
```
Comment (simeng-li) [inline packages/core/src/queries/user.ts:102]: @logto-io/eng 
Should we filter the admin role at the query level?  As the query method name itself does not indicate it's a non-admin user query.

Comment (wangsijie) [reply packages/core/src/queries/user.ts]: Now provides a filter option `hideAdminUser`, PTAL

File: packages/core/src/queries/user.ts [hunk 2]
```diff
@@ -94,19 +94,37 @@ const buildUserSearchConditionSql = (search: string) => {
...
 };
 
-export const countUsers = async (search?: string) =>
+const buildUserConditions = (search?: string, hideAdminUser?: boolean) => {
+  if (hideAdminUser) {
+    return sql`
+      where not (${fields.roleNames}::jsonb?${UserRole.Admin})
+      ${conditionalSql(search, (search) => sql`and (${buildUserSearchConditionSql(search)})`)}
+    `;
+  }
```
Comment (demonzoo) [inline packages/core/src/queries/user.ts:103]: nit: add test cases to cover this part

**Review by demonzoo — APPROVED** [review]

**Review by IceHe — APPROVED** [review]
Overall LGTM

File: packages/core/src/queries/user.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/queries/user.ts:103]: added

**Pull request merged**